### PR TITLE
CB-6347 Redbeams autosync for start/stop

### PR DESCRIPTION
**Pull request by pkedvessy**
CB-6347 Redbeams autosync for start/stop

### CODE REVIEW

File: redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/termination/action/RedbeamsTerminationActions.java [hunk 1]
```diff
@@ -99,6 +103,8 @@ protected Selectable createRequest(RedbeamsContext context) {
                 // through the termination
                 metricService.incrementMetricCounter(MetricType.DB_TERMINATION_FINISHED, context.getDBStack());
                 dbStackService.delete(context.getDBStack());
+                dbStackJobService.unschedule(context.getDBStack());
```
Comment (topolyai5) [inline redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/termination/action/RedbeamsTerminationActions.java:106]: You should call the `unschedule` at the beginning of the termination. As I see, the termination statuses are not the part of the `getAutoSyncStatuses()`

File: redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private DBStackStatusUpdater dbStackStatusUpdater;
+
+    public void sync(DBStack dbStack) {
+        DetailedDBStackStatus detailedDBStackStatus = getDetailedDBStackStatusFromProvider(dbStack);
+        Status status = detailedDBStackStatus.getStatus();
```
Comment (topolyai5) [inline redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java:47]: if any error is occurring at `External DB status lookup failed`, you will get an NPE

File: redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package com.sequenceiq.redbeams.sync;
+
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.statuschecker.service.JobService;
+import org.junit.Before;
```
Comment (topolyai5) [inline redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java:5]: please use junit from jupiter package

File: redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        switch (externalDatabaseStatus) {
+            case STARTED: detailedDBStackStatus = DetailedDBStackStatus.STARTED;
+                break;
+            case STOPPED: detailedDBStackStatus = DetailedDBStackStatus.STOPPED;
+                break;
+            case STOP_IN_PROGRESS: detailedDBStackStatus = DetailedDBStackStatus.STOP_IN_PROGRESS;
+                break;
+            case START_IN_PROGRESS: detailedDBStackStatus = DetailedDBStackStatus.START_IN_PROGRESS;
+                break;
+            default: detailedDBStackStatus = DetailedDBStackStatus.UNKNOWN;
```
Comment (topolyai5) [inline redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java:67]: is `STARTED ` means `AVAILABLE`? What is the difference between `STARTED` and `AVAILABLE`?

File: redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java (hunk 3 above)
Comment (pkedvessy) [reply redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java:5]: This module is using Junit 4

File: redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java (hunk 4 above)
Comment (pkedvessy) [reply redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java:67]: Yes, STARTED means AVAILABLE but in a different context. As far as I can see AVAILABLE is used by legacy reasons.

File: redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java (hunk 2 above)
Comment (pkedvessy) [reply redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java:47]: Good catch

File: redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java (hunk 3 above)
Comment (topolyai5) [reply redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java:5]: I've checked and the jupiter api is added to the module

File: redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java (hunk 4 above)
Comment (topolyai5) [reply redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java]: if in STARTED state, the ui show the db as available?

Comment (pkedvessy) [reply redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java]: I have not created new status messages for the UI so if it is shown up "available" after the creation, it should be "available" when the start operation is finished.

File: redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java (hunk 3 above)
Comment (pkedvessy) [reply redbeams/src/test/java/com/sequenceiq/redbeams/sync/DBStackJobServiceTest.java:5]: Ahh, I see, this module contains a mixture of Junit 4 and 5 tests.

File: redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java (hunk 4 above)
Comment (pkedvessy) [reply redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java]: Where do we display the status on the UI?
Anyway, the status in the DB is AVAILABLE both after start and create, but after start the detaileddbstackstatus is STARTED.

Comment (topolyai5) [reply redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java]: Oh, I see.

**Pull request merged**